### PR TITLE
First part - basic performance test support

### PR DESCRIPTION
**Pull request by ChGen**
First part squashed - basic performance test support: testing scripts, configs and test data.
This PR provides performance testing for uploading & downloading for `maven` and `files` adapter for the Artipie. Tests use Apache JMeter and Docker. Artipie runs in Docker container and relies on local filesystem storage.
Output is directory in format: `files_dl_${host}_${port}_${duration}_$(date +%y-%m-%d_%H-%M-%S)`.  It contains requests log, html dashboard with results and `statistics.json` for parsing or CLI use.
- `jmeterReport.sh` allows to generate report files for given JMeter jmx file.
- `Dockerfile.client` allows to run for recording `conan install` command with given Conan server URL and JMeter proxy URL
- `artipie-docker.sh` run test docker instance of artipie
- `jmx-files-dl.sh` download files by list, generated for upload script above
- `jmx-files-ul.sh` upload random binary data files, like `jmx-run.sh`
- `jmx-files-maven-ul.sh` upload maven artifacts test set to the files repository adapter (bintest repo)
- `jmx-files-maven-dl.sh` download maven artifacts test set to the files repository adapter (bintest repo)
- `jmx-run.sh` binary data test with generated random data of small/medium/large files
- `jmx-local.sh` run local artipie + binary test (above), then stop artipie
- `jmx-maven-ul.s` upload maven artifacts test set to the maven repo (maventest repo)
- `jmx-maven-dl.s` download maven artifacts test set from the maven repo (maventest repo)
- `maven-repo-reset.sh` reset & wipe data for maven repository via Artipie REST API, for tests
- `prep-maven.sh` downloads required maven artifact files for tests. Results put in `upload-maven-src/repository/`
- `loadtests/helloworld-src` simplest maven project for test purposes
- `loadtests/*.jmx` JMeter files, which are used from the shell scripts before
- `loadtests/*.csv` list of files for download/upload testing
- `loadtests/root/*` artipie repository directory structure with required config files

### CODE REVIEW

**Comment by ChGen** [comment]
Monitoring will be squashed as the next PR. 
Updated docs & old test data cleanup will follow.

**Comment by ChGen** [comment]
@dgarus , please review.
Monitoring will be squashed as the next PR.
Updated docs & old test data cleanup will follow.

**Comment by ChGen** [comment]
@andpopov @Swizbiz , please review.

**Comment by dgarus** [comment]
> @andpopov @Swizbiz , please review.

@ChGen Too many files!
You have to make detailed PR description and write what were done, directory structure and so on which may help do the review.

**Comment by Swizbiz** [comment]
@ChGen 
I think, that we don't need push all maven files in `upload-maven-src`. Dependencies should be downloaded and this folder should be filled in compile/build stage.

**Comment by ChGen** [comment]
@Swizbiz , I decided to put files set in the repository, because a lot of the maven dependencies are indirect, but I want actual files to be exactly the same all the time, so we could make comparisons between various runs and even in disconnected/restricted environments.  And make it simpler/more predictable on runs too.

**Comment by dgarus** [comment]
> @Swizbiz , I decided to put files set in the repository, because a lot of the maven dependencies are indirect, but I want actual files to be exactly the same all the time, so we could make comparisons between various runs and even in disconnected/restricted environments. And make it simpler/more predictable on runs too.

But we can just generate files which we need for testing, so I agree with @Swizbiz.

**Comment by ChGen** [comment]
@dgarus , as I understand, dependencies may be potentially changed on the remote repositories (e.g. change latest version number or critical fix), which will make file set to be different. Local copy provides us stable test data set, which is the priority for performance tests.

**Comment by dgarus** [comment]
> @dgarus , as I understand, dependencies may be potentially changed on the remote repositories (e.g. change latest version number or critical fix), which will make file set to be different. Local copy provides us stable test data set, which is the priority for performance tests.

No, I don't know cases like this. Every bug fix is in new version. But I say that we could **generate** files which we need to do tests.

**Comment by ChGen** [comment]
@dgarus , maven adapter checks that files are correct and not broken, so random data files won't be enough for it. We could actually make some maven test data generator, but as a separate next step, since it will have some complexity.

**Comment by dgarus** [comment]
> @dgarus , maven adapter checks that files are correct and not broken, so random data files won't be enough for it. We could actually make some maven test data generator, but as a separate next step, since it will have some complexity.

I don't think it is too hard. You can create a simple jar file with custom resource.

**Comment by ChGen** [comment]
@dgarus , I will also have to generate metadata files, and in the end, it will be just repetitive upload/download of the uniform, almost the same, artifacts. But it could be useful as an extra set of tests, in addition to the current ones. This maven repository test set provides full set of artifacts of basic maven project with real metadata and jars, which is more similar to typical maven workload.

**Comment by ChGen** [comment]
So, we decided that current files set should be removed and will be downloaded via maven with script, and separate issue will be created for implementing configurable maven test data generation.

**Comment by ChGen** [comment]
@dgarus , please review, files set is removed and must be downloaded and prepared via separate script `loadtests/prep-maven.sh`

**Comment by ChGen** [comment]
@olenagerasimova , please review.

File: loadtests/root/var/repo/_api_permissions.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+endpoints:
```
Comment (olenagerasimova) [inline loadtests/root/var/repo/_api_permissions.yaml]: this file is not required and not used by artipie, let's remove it

File: loadtests/root/var/repo/_permissions.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+permissions:
```
Comment (olenagerasimova) [inline loadtests/root/var/repo/_permissions.yaml]: this one too

File: loadtests/maven-repo-reset.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    "repo": {
+    "type": "maven",
+    "storage": "default",
+    "permissions": {
+      "*": [
+        "download",
+        "upload",
+        "write",
+        "publish",
+        "deploy"
```
Comment (olenagerasimova) [inline loadtests/maven-repo-reset.sh]: maven repo in artipie supports just upload and download

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@ChGen thanks, my point here - let's remove all the extra files (not used repo configs too) to keep artipie configuration as minimal and simple as possible

**Comment by ChGen** [comment]
@olenagerasimova , please review, I fixed review issues.

**Review by olenagerasimova — APPROVED** [review]
@ChGen thanks

**Comment by ChGen** [comment]
@dgarus  , should we merge this?

**Review by dgarus — APPROVED** [review]

**Pull request merged**